SmartDeviceLink/SDLProtocol.m

```diff
@@ -203,7 +203,7 @@ - (BOOL)sendRPC:(SDLRPCMessage *)message encrypted:(BOOL)encryption error:(NSErr
             // Build a binary header
             // Serialize the RPC data into an NSData
             SDLRPCPayload *rpcPayload = [[SDLRPCPayload alloc] init];
-            rpcPayload.functionID = [[[[SDLFunctionID alloc] init] getFunctionID:[message getFunctionName]] intValue];
+            rpcPayload.functionID = [[[SDLFunctionID sharedInstance] functionIdForName:[message functionNameForId]] intValue];
```

Comment:
A few things about this seem odd.

1. Why are we getting the name for the id, then the id for the name? Seems circular?

2. Wouldn't the message's `functionNameForId` method just be `functionName` if we're not passing anything? Can't find if or where that changed in this PR though.